### ISSUE TITLE
Spurious undef variable on assignment inside if statement

### ISSUE THREAD

**Issue by KristofferC** [JuliaEditorSupport/julia-vscode#234]
![image](https://user-images.githubusercontent.com/1282691/28922205-767502ac-7859-11e7-9b65-f89d69e241ea.png)

`HELLO` is defined here

**Comment by ZacLN** [comment]
Hmm that's a complicated one, I'm not quite sure what we should do in the case of
```
(x = 1) == y || (z =1)
```
is `z` defined or not?

I'm going to leave this for later

**Comment by KristofferC** [comment]
Yeah, the dynamical nature of Julia means that sometimes you won't know if a variable will be defined without running the code. But in this case,  there are no conditional definitions of variables like in your example. `HELLO` will always be defined and I think the code here is equivalent to

```
HELLO = "str"
if HELLO == "str
    print("hello")
end
```
